implementation/service_discovery/src/service_discovery_impl.cpp

```diff
@@ -3235,6 +3236,13 @@ service_discovery_impl::start_main_phase_timer() {
                     this, std::placeholders::_1));
 }
 
+void
+service_discovery_impl::stop_main_phase_timer() {
+    std::lock_guard<std::mutex> its_lock(main_phase_timer_mutex_);
```

Comment:
As suggested by sonarcube
```suggestion
    std::scoped_lock<std::mutex> its_lock(main_phase_timer_mutex_);
```

Reply:
@kheaactua what do you think of this change?

Reply:
I don't think it adds anything....  Reading up on this, when dealing with a single mutex `lock_guard` seems equivalent.  There are arguments on SO where some people try to claim `lock_guard` is deprecated, but I see no evidence of that in the standard.

I suppose I don't have strong feelings either way. :)

Reply:
@kheaactua could you commit this change so that this is merged?

Reply:
Done!  I used a fixup commit so it can be easily squashed.